google/cloud/apigeeregistry/v1/apihub/fields.proto

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// (-- api-linter: core::0215::versioned-packages=disabled
+//     aip.dev/not-precedent: Support protos for the apigeeregistry.v1 API. --)
+package google.cloud.apigeeregistry.v1.apihub;
+
+import "google/api/field_behavior.proto";
+
+option java_package = "com.google.cloud.apigeeregistry.v1.apihub";
+option java_multiple_files = true;
+option java_outer_classname = "FieldSetProto";
+option go_package = "github.com/apigee/registry/pkg/application/apihub;apihub";
+
+// Defines a structure for general field storage.
+message FieldSet {
+  // Artifact identifier. May be used in YAML representations to indicate the id
+  // to be used to attach the artifact.
+  string id = 1;
+
+  // Artifact kind. May be used in YAML representations to identify the type of
+  // this artifact.
+  string kind = 2;
+
+  // A short displayable name for the field set.
+  string display_name = 3;
+
+  // A description of the field set.
+  string description = 4;
+
+  // Full resource name of an FieldSetDefinition artifact that
+  // describes this field set.
+  string definition_name = 5 [(google.api.field_behavior) = REQUIRED];
+
+  // The field values, stored using field ids as keys.
+  map<string,string> values = 6;
```

Comment:
Is a field set essentially a display grouping? If so, should we consider separating the display metadata from the data?

Reply:
There is display metadata in the FieldSetDefinition - if you mean that the grouping itself is metadata, that seems to get burdensome for applications (viewers) that would have to fetch all these things individually. But we might consider this like a ScoreCard that is derived from individual artifacts.

However, the overhead of storing individual fields as separate artifacts is pretty high, so storing fields in groups seems more practical.

I like to think of these field sets as conceptually-related groups of fields, but that's up to applications and users to eventually decide.

Reply:
I guess what's confusing me is the `display_name` and `description` fields here.

Reply:
Are these just for convenience in browsing artifacts?

Reply:
Maybe it's overkill, but we have these fields for Scores, which is somewhat similar: https://github.com/apigee/registry/blob/a2537feb98ba82b98607d6a9e14519a1455afb3f/google/cloud/apigeeregistry/v1/scoring/score.proto#L41

They are instance-specific values, which, at scale, we probably won't want to be using.

Reply:
I've removed display_name and description from FieldSet. We can add them later if we find a need for them.